lib/runtime/router.ts

```diff
@@ -251,6 +251,28 @@ export default class Router extends DenaliObject implements RouterDSL {
   }
 
   /**
+   * Returns the URL for a given action. You can supply a params object which
+   * will be used to fill in the dynamic segements of the action's route (if
+   * any).
+   */
+  urlFor(action: string | Action, data: Object): string | null {
+    if (typeof action === 'string') {
+      action = this.container.lookup(`action:${ action }`);
+    }
+    if (!action) {
+      return null;
+    }
+
+    let route: Route;
+    forEach(this.routes, (routes) => {
+      route = find(routes, { action: <Action> action });
+      return !route; // kill the iterator if we found the match
+    });
+
+    return route ? (<string> route.reverse(data)) || null : null;
```

Comment:
definitely - I mainly just copied over the original method implementation from denali 0.0.20 - tried to leave the behavior as intact as possible.